ql/test/library-tests/semmle/go/frameworks/StdlibTaintFlow/ArchiveTar.go

```diff
@@ -0,0 +1,105 @@
+// WARNING: This file was automatically generated. DO NOT EDIT.
+
+package main
+
+import (
+	"archive/tar"
+	"io"
+	"os"
+)
+
+func TaintStepTest_ArchiveTarFileInfoHeader_B0I0O0(sourceCQL interface{}) interface{} {
+	fromFileInfo656 := sourceCQL.(os.FileInfo)
+	intoHeader414, _ := tar.FileInfoHeader(fromFileInfo656, "")
+	return intoHeader414
+}
+
+func TaintStepTest_ArchiveTarNewReader_B0I0O0(sourceCQL interface{}) interface{} {
+	fromReader518 := sourceCQL.(io.Reader)
+	intoReader650 := tar.NewReader(fromReader518)
+	return intoReader650
+}
+
+func TaintStepTest_ArchiveTarNewWriter_B0I0O0(sourceCQL interface{}) interface{} {
+	fromWriter784 := sourceCQL.(*tar.Writer)
+	var intoWriter957 io.Writer
+	intermediateCQL := tar.NewWriter(intoWriter957)
+	link(fromWriter784, intermediateCQL)
+	return intoWriter957
+}
+
+func TaintStepTest_ArchiveTarHeaderFileInfo_B0I0O0(sourceCQL interface{}) interface{} {
+	fromHeader520 := sourceCQL.(tar.Header)
+	intoFileInfo443 := fromHeader520.FileInfo()
+	return intoFileInfo443
+}
+
+func TaintStepTest_ArchiveTarReaderNext_B0I0O0(sourceCQL interface{}) interface{} {
+	fromReader127 := sourceCQL.(tar.Reader)
+	intoHeader483, _ := fromReader127.Next()
+	return intoHeader483
+}
+
+func TaintStepTest_ArchiveTarReaderRead_B0I0O0(sourceCQL interface{}) interface{} {
+	fromReader989 := sourceCQL.(tar.Reader)
+	var intoByte982 []byte
+	fromReader989.Read(intoByte982)
+	return intoByte982
+}
+
+func TaintStepTest_ArchiveTarWriterWrite_B0I0O0(sourceCQL interface{}) interface{} {
+	fromByte417 := sourceCQL.([]byte)
+	var intoWriter584 tar.Writer
+	intoWriter584.Write(fromByte417)
+	return intoWriter584
+}
+
+func TaintStepTest_ArchiveTarWriterWriteHeader_B0I0O0(sourceCQL interface{}) interface{} {
+	fromHeader991 := sourceCQL.(*tar.Header)
+	var intoWriter881 tar.Writer
+	intoWriter881.WriteHeader(fromHeader991)
+	return intoWriter881
+}
+
+func RunAllTaints_ArchiveTar() {
+	{
+		source := newSource(0)
+		out := TaintStepTest_ArchiveTarFileInfoHeader_B0I0O0(source)
+		sink(0, out)
```

Comment:
I think it would make the tests cleaner to just inline these functions; is there a reason you haven't?

Reply:
What do you mean, exactly?

Reply:
Why do you have separate functions for each step? It seems like you could just write the step itself into these blocks, like so:

```go
{
    source = newSource(0)
    var intoWriter881 tar.Writer
    intoWriter881.WriteHeader(source)
    sink(intoWriter881)
}
```

It would be much more compact and probably more informative when a test fails.

Reply:
That's definitely possible, but I prefer named functions; they keep things clean and ordered, and I can easily understand where and for what each function was generated (see the naming suffix).

Also, less visual fatigue in case of a review, not to mention a lot less scrolling in case of a 100+ test cases.

Reply:
I'll draft an implentation of that tomorrow, and share the results of what it looks like.

Reply:
Are you still looking into this? I'd be fine with leaving the tests as is (though I don't necessarily agree with your reasoning; the generated function names do not look very pretty, and I find a list of 100+ functions at least as tiresome to review as a list of 100+ code blocks).

Reply:
Alternative suggestion:

```suggestion
		sink(0, TaintStepTest_ArchiveTarFileInfoHeader_B0I0O0(newSource(0))
```

Reply:
You might then even be able to rename `newSource` to `source`, which would be nicely parallel to `sink`.